### DISCUSSION TITLE
How to exlude directories from source

### DISCUSSION THREAD

**Discussion by sebastianmacarescu** [llaville/php-compat-info#304] [Q&A]
Hi all,

I saw that the json config file was excluded in favor of php config files as written here:
https://github.com/llaville/php-compat-info/tree/master/docs/02_Configs

The issue is I can't really find an example on how to exclude the vendor dir for example. Does anybody have an example on how I can achieve that?

**Comment by llaville** [comment]
Hi Sebastian,

If you want to exclude vendor directory from data source to parse, you should : 
- if using the CLI version, specify only the directory that host source files, and not the root folder of the project `bin/phpcompatinfo analyser:run src/ `
- if using API version (https://github.com/llaville/php-compat-info/blob/master/examples/api_analyser_run.php#L23), specify at line 23 only the directory to scan.

In lot of projects, directories are structured like that : 

```
root_folder
|
+-- src/
|
+--tests/
| 
+--vendor/
```
Hope I've answered to your question

**Reply by sebastianmacarescu** [reply]
Hi and thank you for your response
The repo structure is (laravel):
* app
* bootstrap
* config
* database
* some-other-lib
* vendor
* tests

I know that I can scan only app for the source code but I need to scan everything except the vendor and tests.  For example I have php code in config, database, some-other-lib etc.
Also I'm using this tool in CI to check for required extensions.

The alternative would be to list all directories and run the tool for each one of them if it's not vendor or tests. Then do some scripting to merge the outputs.

I looked on the source code and providing an exclude option shouldn't be too hard.

**Reply by llaville** [reply]
Thanks for your feedback. I'm agree to add a `exclude` option to restrict data source to analyse.

**Reply by sebastianmacarescu** [reply]
Ty.

**Comment by llaville** [comment]
Use the new `--exclude` option introduced in version 5.5.2 (see feature request https://github.com/llaville/php-compat-info/issues/305), and documentation update at https://github.com/llaville/php-compat-info/blob/5.5/docs/04_Exclude_Folders/README.md

**Reply by sebastianmacarescu** [reply]
Thank you very much :)
